s3-sanity-test.sh

```diff
@@ -75,24 +75,46 @@ Operations performed:
   * Delete User
   * Delete Account"
 
-cleanup() {
+update_config() {
+
+   if [ ! -z $end_point ];then
+     s3iamcli listaccounts --ldapuser sgiamadmin --ldappasswd "$ldappasswd" >/dev/null 2>&1 || echo "configured s3iamcli"
+     echo "using s3endpoint $end_point"
+     ls /root/.sgs3iamcli/config.yaml 2> /dev/null || { echo "S3iamcli configuration file is missing" && exit 1; }
+     ls /root/.s3cfg 2> /dev/null || { echo "S3cmd configuration file is missing" && exit 1; }
+     sed -i "s/IAM:.*/IAM: http:\/\/$end_point:9080/g" /root/.sgs3iamcli/config.yaml
+     sed -i "s/IAM_HTTPS:.*/IAM_HTTPS: https:\/\/$end_point:9443/g" /root/.sgs3iamcli/config.yaml
+     sed -i "s/VERIFY_SSL_CERT:.*/VERIFY_SSL_CERT: false/g" /root/.sgs3iamcli/config.yaml
+     sed -i "s/host_base =.*/host_base = $end_point/g" /root/.s3cfg
+   fi
 
-    if [ "$externalcleanup" = true ]
-    then
-        output=$(s3iamcli resetaccountaccesskey -n SanityAccountToDeleteAfterUse --ldapuser sgiamadmin --ldappasswd "$ldappasswd")
-        echo $output
-        access_key=$(echo -e "$output" | tr ',' '\n' | grep "AccessKeyId" | awk '{print $3}')
-        secret_key=$(echo -e "$output" | tr ',' '\n' | grep "SecretKey" | awk '{print $3}')
-        TEST_CMD="s3cmd --access_key=$access_key --secret_key=$secret_key"
-    fi
-
-    $TEST_CMD del "s3://sanitybucket/SanityObjectToDeleteAfterUse" || echo "Failed"
-    $TEST_CMD rb "s3://sanitybucket" || echo "Failed"
-    s3iamcli deleteuser -n SanityUserToDeleteAfterUse --access_key $access_key --secret_key $secret_key || echo "Failed"
-    s3iamcli deleteaccount -n SanityAccountToDeleteAfterUse --access_key $access_key --secret_key $secret_key || echo "Failed"
-    rm -f $test_file_input $test_output_file || echo "failed"
-    sed -i "s/VERIFY_SSL_CERT:.*/VERIFY_SSL_CERT: True/g" /root/.sgs3iamcli/config.yaml
-    exit 0
+}
+
+restore_config() {
+   ls /root/.sgs3iamcli/config.yaml 2> /dev/null || { echo "S3iamcli configuration file is missing" && exit 1; }
+   ls /root/.s3cfg 2> /dev/null || { echo "S3cmd configuration file is missing" && exit 1; }
+   rm -rf /root/.sgss3iamcli
+   sed -i "s/host_base =.*/host_base = s3.seagate.com/g" /root/.s3cfg
+}
+
+cleanup() {
+   update_config
+   output=$(s3iamcli resetaccountaccesskey -n SanityAccountToDeleteAfterUse --ldapuser sgiamadmin --ldappasswd "$ldappasswd") || echo ""
+   if [[ $output == *"NoSuchEntity"* ]]
+   then
+      echo "Sanity Account doesn't exist"
+      return;
+   else
+      access_key=$(echo -e "$output" | tr ',' '\n' | grep "AccessKeyId" | awk '{print $3}')
+      secret_key=$(echo -e "$output" | tr ',' '\n' | grep "SecretKey" | awk '{print $3}')
+      TEST_CMD="s3cmd --access_key=$access_key --secret_key=$secret_key"
+      $TEST_CMD del "s3://sanitybucket/SanityObjectToDeleteAfterUse" || echo "Failed"
```

Comment:
please update echo messages with proper messages like "Failed to delete object "objectName" using s3cmd"